waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/MetaStoreMapping.java

```diff
@@ -51,14 +51,14 @@
   MetaStoreMapping checkWritePermissions(String databaseName);
 
   /**
-   * @param database (assummed to be the database used in this mapped metastore so any waggle dance related prefix must
+   * @param database (assumed to be the database used in this mapped metastore so any waggle dance related prefix must
    *          have been stripped already)
-   * @return throws {@link NotAllowedException} when the metastore mapped by this class does not have the correct
-   *         permission otherwise calls {{@link #getClient().create_database(database)}
-   * @throws TException
-   * @throws MetaException
-   * @throws InvalidObjectException
-   * @throws AlreadyExistsException
+   * @throws NotAllowedException when the metastore mapped by this class does not have the correct permission otherwise
+   *           calls getClient().create_database(database)
+   * @throws TException Thrift exception
```

Comment:
Usually the description is what can lead to the exception being thrown, like you've done on line 61. It's not that straightforward to actually find the reasons as Hive's own documentation doesn't include them :( so I'm just going to put what I think the best we can do here is. For this one - "if a Thrift error occurs".